lib/json/serialization.nit

```diff
@@ -461,25 +472,34 @@ class JsonDeserializer
 	#     serialize
 	#
 	#     var error: String
+	#     var related_data: MyData
 	# end
 	#
 	# class MyJsonDeserializer
 	#     super JsonDeserializer
 	#
 	#     redef fun class_name_heuristic(json_object)
```

Comment:
It was my first version, but it also broke the clients and provided less versatility for complex cases. I don't think it will be too much of an issue in practice because you can always group the checks, or add a service method as needed. Plus, my next PR gives access to the static type, it should make this pretty much useless in most cases.